modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4ChunkedHttpContinuation.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http.netty4;
+
+import io.netty.util.concurrent.PromiseCombiner;
+
+import org.elasticsearch.rest.ChunkedRestResponseBody;
+
+public final class Netty4ChunkedHttpContinuation implements Netty4HttpResponse {
```

Comment:
I think we don't want this class to be public since it's fully managed within `Netty4HttpPipeliningHandler`. Maybe it could even be an inner class.

Reply:
It can't be a private inner class because we need to mention it when defining the sealed `Netty4HttpResponse` interface. But we can make it package-private, see 708fa524b95.

Reply:
Yeah I was thinking a package private class with a private constructor.

Reply:
> with a private constructor

A package-private constructor right? A fully private constructor wouldn't be accessible where it's needed. Not sure if I'm missing something here.

Reply:
I meant an inner class of `Netty4HttpPipeliningHandler`, e.g.:
```
public class Netty4HttpPipeliningHandler {
    final class Netty4ChunkedHttpContinuation implements Netty4HttpResponse { 
        private Netty4ChunkedHttpContinuation() { }
    }
}

public sealed interface Netty4HttpResponse permits Netty4FullHttpResponse, Netty4ChunkedHttpResponse, Netty4HttpPipeliningHandler.Netty4ChunkedHttpContinuation {
}
```